packages/next/src/lib/eslint/runLintCheck.ts

```diff
@@ -369,8 +369,16 @@ export async function runLintCheck(
         } else {
           // Check if necessary deps installed, and install any that are missing
           deps = await hasNecessaryDependencies(baseDir, requiredPackages)
-          if (deps.missing.length > 0)
+          if (deps.missing.length > 0) {
+            deps.missing.forEach(({ pkg }) => {
+              if (pkg === 'eslint') {
+                // eslint v9 has breaking changes, so lock to 8 until dependency plugins fully support v9.
+                pkg = 'eslint@^8'
+              }
+            })
```

Comment:
This is performing a local mutation and not actually updating the `pkg` property of the dependency. I think you maybe meant to do something like this
```suggestion
          if (deps.missing.length > 0) {
            deps.missing.forEach((dep) => {
              if (dep.pkg === 'eslint') {
                // eslint v9 has breaking changes, so lock to 8 until dependency plugins fully support v9.
                dep.pkg = 'eslint@^8'
              }
            })
```

Reply:
Thanks for catching that!